packages/hash/shared/src/prosemirror.ts

```diff
@@ -105,6 +19,203 @@ export const isEntityNode = (
 
 export const componentNodeGroupName = "componentNode";
 
+export const createSchema = () =>
+  new Schema({
+    nodes: {
+      doc: {
+        content: "((componentNode|block)+)|blank",
+      },
+      blank: {
+        /**
+         * As we don't have any component nodes defined by default, we need a
+         * placeholder, otherwise Prosemirror will crash when trying to
+         * interpret the content expressions in other nodes. However, as soon
+         * as we have defined a different component node, we remove the blank
+         * node from the componentNode group, which ensures that when
+         * Prosemirror attempts to instantiate a componentNode it uses that
+         * node instead of the blank one
+         *
+         * @see import("./ProsemirrorManager.ts").ProsemirrorManager#prepareToDisableBlankDefaultComponentNode
+         */
+        group: componentNodeGroupName,
+        toDOM: () => ["div", 0] as const,
+      },
+      block: {
+        content: "entity",
+        /**
+         * These properties are necessary for copy and paste (which is
+         * necessary for drag and drop)
+         */
+        toDOM: () => {
+          return [
+            "div",
+            {
+              // @todo this isn't applied because of the node view
+              "data-hash-type": "block",
+            },
+            0,
+          ] as const;
+        },
+        parseDOM: [
+          {
+            // @todo is this necessary
+            tag: 'div[data-hash-type="block"]',
```

Comment:
Similar comment to just above – if this is not something you'll be addressing in a follow up, can we add a bit more context?

Reply:
Updated the comment [`2ce9352` (#872)](https://github.com/hashintel/hash/pull/872/commits/2ce9352e6f0a73d91a2963f3e458a09fba702f03)